anndata/tests/test_concatenate.py

```diff
@@ -1018,6 +1019,15 @@ def gen_something(n):
     return np.random.choice(options)(n)
 
 
+def gen_3d_numeric_array(n):
+    return np.random.randn(n, n, n)
+
+
+def gen_3d_recarray(_):
+    # Ignoring n as it can get quite slow
```

Comment:
What does this mean exactly?

Reply:
Which part?

`recarray` refers to a record array: https://numpy.org/doc/stable/reference/generated/numpy.recarray.html#numpy-recarray

n is a parameter passed to all the other functions here. Generally it changes the size of the result, but we ignore that in a couple cases. Here I ignore it because it made some test cases take over a minute.